tad_pathways/tad.py

```diff
@@ -0,0 +1,189 @@
+        )
+
+    def build_snp_list(self):
+        command_list = [
+            "Rscript",
+            "--vanilla",
+            "scripts/build_snp_list.R",
+            "--snp_file",
+            self.snp_list_location,
+            "--output_file",
+            self.snp_location_file,
+        ]
+        subprocess.call(command_list)
+
+    def build_custom_tad_genelist(self):
+        command_list = [
+            "python",
+            "scripts/build_custom_tad_genelist.py",
+            "--snp_data_file",
+            self.snp_location_file,
+            "--output_file",
+            self.genelist_file,
+        ]
+
+        if self.remove_hla:
+            command_list += ["--remove_hla_tad"]
+
+        subprocess.call(command_list)
+
+    def webgestalt_run(self):
+        command_list = [
+            "Rscript",
+            "--vanilla",
+            "scripts/webgestalt_run.R",
+            "--tad_genelist_file",
+            self.genelist_file,
+            "--output_name",
+            self.snp_list_name,
+            "--output_directory",
+            self.base_dir,
+        ]
+        subprocess.call(command_list)
+
+    def check_webgestalt(self):
+        output_pval_file = os.path.join(
+            self.base_dir, "{}_pvals.tsv".format(self.snp_list_name)
+        )
+        if os.path.exists(output_pval_file):
+            return True
+        else:
+            return False
+
+    def get_evidence(self):
+        # The first command builds the evidence
+        command_list = [
+            "python",
+            "scripts/construct_evidence.py",
+            "--trait",
+            self.snp_list_name,
+            "--gwas",
+            self.nearest_gene_file,
+            "--pathway",
+            self.pathway_p_values_file,
+            "--results_directory",
+            self.base_dir,
+            "--gestalt_directory",
+            self.base_dir,
+        ]
+        if self.all_pathways:
+            command_list += ["--all_pathways"]
+            evidence_file = self.all_pathways_evidence
+            output_file = self.all_pathways_summary
+        else:
+            evidence_file = self.gene_evidence_file
+            output_file = self.evidence_summary_file
+
+        subprocess.call(command_list)
+
+        # The second command summarizes this evidence
+        command_list = [
+            "python",
+            "scripts/summarize_evidence.py",
+            "--evidence",
+            evidence_file,
+            "--snps",
+            self.genelist_file,
+            "--output_file",
+            output_file,
+        ]
+        subprocess.call(command_list)
+
+    def compile_results(self):
+        self.build_snp_list()
+        self.build_custom_tad_genelist()
+        self.webgestalt_run()
+        webgestalt_exists = self.check_webgestalt()
+        if webgestalt_exists:
+            self.get_evidence()
+            return 1
```

Comment:
Just curious, the return value of this method will be used later?

Reply:
Hmm, doesn't look like I use this.. Not sure why its here. Maybe some functionality that I decided not to use. I will remove for now